webapp/src/screens/tabs/UsersTable.tsx

```diff
@@ -0,0 +1,923 @@
+            password: password ? Buffer.from(password, "ascii").toString("base64") : ""
+          },
+          roleNames: roleNames ? roleNames.split(",") : [],
+          courseIds: courseIds ? courseIds.split(",") : [],
+          photoUrl: photoUrl ? photoUrl : "",
+        }
+        try {
+          const successful = await user === undefined ? createNewUser(userPartial) : updateUser(userPartial);
+          if (!successful) {
+            setNewUserCreationError(true);
+          }
+          setPassword('');
+          handleClose();
+        } catch (e) {
+          setNewUserCreationError(true);
+        }
+      }
+    })();
+  }, [user, courseIds, lastName, roleNames, firstName, photoUrl, emailAddress, username, password])
+
+  const openDialog = useMemo(() => openNewUserDialog, [openNewUserDialog]);
+
+  const handleFirstNameChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setFirstNameError(false);
+    setFirstName(e.target.value);
+  }, [firstName, firstNameError]);
+
+  const handleLastNameChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setLastNameError(false);
+    setLastName(e.target.value);
+  }, [lastName, lastNameError]);
+
+  const handleEmailAddressChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmailAddressError(false);
+    setEmailAddress(e.target.value);
+  }, [emailAddress, emailAddressError]);
+
+  const handleUsernameChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setUsernameError(false);
+    setUsername(e.target.value);
+  }, [username, usernameError]);
+
+  const handlePasswordChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setPasswordError(false);
+    setPassword(e.target.value);
+  }, [password, passwordError]);
+
+  const handleRoleNamesChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setRoleNamesError(false);
+    setRoleNames(e.target.value);
+  }, [roleNames, roleNamesError]);
+
+  const handleCourseIdChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setCourseIdsError(false);
+    setCourseIds(e.target.value);
+  }, [courseIds, courseIdsError]);
+
+  const handlePhotoUrlChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setPhotoUrlError(false);
+    setPhotoUrl(e.target.value);
+  }, [photoUrl, photoUrlError]);
+
+  useEffect(() => {
+    if (user !== null && user !== undefined) {
+      setFirstName(user.firstName);
+      setLastName(user.lastName);
+      setEmailAddress(user.emailAddress);
+      setUsername(user.username);
+      setCourseIds(user.courseIds ? user.courseIds.join(',') : "");
+      setRoleNames(user.roleNames.join(','));
+      setPhotoUrl(user.photoUrl);
+    } else if (user === null || user === undefined) {
+      setFirstName("");
+      setLastName("");
+      setEmailAddress("");
+      setUsername("");
+      setCourseIds("");
+      setRoleNames("");
+      setPhotoUrl("");
```

Comment:
Ensure that you set password to "" as well otherwise the password field keeps its text